components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -41,6 +50,22 @@ class DownloadsFeature(
      */
     fun start() {
         super.observeSelected()
+        fun reAttachOnStartDownloadListener(previousDialog: DownloadDialogFragment?) {
+            previousDialog?.apply {
+                this@DownloadsFeature.dialog = this
+                val selectedSession = sessionManager.selectedSession
+                val download = selectedSession?.download
+                download?.consume {
```

Comment:
nit: Just chain it? :)

```Kotlin
selectedSession?.download?.consume {
  ...
}
```

Reply:
Chained ⛓ 😆 
https://github.com/mozilla-mobile/android-components/blob/cb34bb9b8d9ceafd5a9d03fe65a7099c5b80efb1/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt#L122